### DISCUSSION TITLE
Where can I find a way to use similar functions through browser login？

### DISCUSSION THREAD

**Discussion by qcgm1978** [Christopher-Hayes/vscode-chatgpt-reborn#41] [General]
### Describe the feature

I want to know where I can find a way to use similar functions through browser login. The problem I face is that the API key prompts me that I have already exceeded the quota, but I can access through the browser, so setting up the key is invalid for me. Is there any version that can be installed locally? I need this function. Thank you for your help.

**Comment by Christopher-Hayes** [comment]
Sorry missed this issue, browser login is not planned.

Browser functionality was part of the original repo before this "reborn" fork, but it was removed before the code was open-sourced. So, this "reborn" fork never had that browser login code to begin with. The original extension developer removed it due to OpenAI ToS violations.

So, this extension has no plans of rebuilding that functionality due to how sophisticated it would have to be to handle OpenAI's anti-bot measures as well as it putting user accounts at risk of getting banned for violating the ToS.

The only alternative I can think of is this extension has a "proxy" config setting for using a proxy server instead of the official OpenAI server. There might be an open-source project that lets you use the browser interface through a local server. I'm not sure if such a project exists, but that's the only possiblity I can think of.

**Reply by Christopher-Hayes** [reply]
I've migrated this issue to a discussion since the feature is not planned, but further discussion about alternatives or work-arounds can take place here.

**Reply by qcgm1978** [reply]
Thanks your reply. I found a vscode extension [ChatGPT - EasyCode](https://marketplace.visualstudio.com/items?itemName=EasyCodeAI.chatgpt-gpt4-gpt3-vscode) that supports to ask chatgpt without key or logging into it although it's not stable enough.
